elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchIndexManagerFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.impl;
+
+import org.hibernate.search.engine.impl.DefaultIndexManagerFactory;
+import org.hibernate.search.indexes.spi.IndexManager;
+
+/**
+ * Index manager capable of resolving the "elasticsearch" alias. All other names/aliases will be handled by the default
+ * implementation.
+ * <p>
+ * Registered through the Java service loader mechanism.
+ *
+ * @author Gunnar Morling
+ */
+public class ElasticsearchIndexManagerFactory extends DefaultIndexManagerFactory {
```

Comment:
@gunnarmorling as explained in the above JIRA, writing it up reminded me of another thing: the Infinispan case goes beyond contributing it actually overrides some names from the default factory. It extends the default to "pass on" all names it doesn't need to override though, so introducing "elasticsearch" in the default would work.